File: 7-kyu/time-degrees/javascript/solution.js
const clockDegree = s => {
  const [hours, minutes] = s.split(":").map(Number);
  if (hours < 0 || hours > 23 || minutes < 0 || minutes > 59 || Number.isNaN(hours) || Number.isNaN(minutes)) {
    return "Check your time !";
  }
  const degreesPerHour = 360 / 12;
  const degreesPerMinute = 360 / 60;
  const hourDegrees = degreesPerHour * (hours === 0 || hours === 12 ? 12 : hours % 12);
  const minuteDegrees = degreesPerMinute * (minutes === 0 ? 60 : minutes);
  return `${hourDegrees}:${minuteDegrees}`;
};

File: 7-kyu/time-degrees/python/solution.py
def clock_degree(s) :
    hh, mm = map(int, s.split(':'))
    if not 0 <= hh <= 23 or not 0 <= mm <= 59:
        return 'Check your time !'
    h_deg = int((360 / 12) * (12 if hh == 0 or hh == 12 else hh % 12))
    m_deg = int((360 / 60) * (60 if mm == 0 else mm))
    return f'{h_deg}:{m_deg}'
